### ISSUE TITLE
Poprawna postać wyświetla się przed zgadnięciem 

### ISSUE THREAD

**Issue by RafalKornel** [dbartoszp/gothicdle#2]
Dzisiaj po północy natrafiłem na błąd: po otwarciu strony (Safari na iPhone, ale raczej nieistotne) przez ułamek sekundy pojawiła mi się dzisiejsza postać, zanim jeszcze ją odgadłem. Przez to niestety miałem "spojler" i nie było sensu odgadywać. 

Steps to reproduce (jak mi się wydaje):
- odgadnij postać w dniu N 
- nie zamykaj strony
- W dniu N+1 wejdź na stronę 

Expected: 
- user widzi tabelkę ze swoimi zgadnięciami oraz poprawną postać (jeśli odgadł) z dnia N

Actual:
- user widzi zgadnięcia z dnia N, ale poprawna postać z dnia N+1

Wydaje mi się że zanim strona zostanie odświeżona (czyli refresh SSR) wywoływane jest query po aktualna postać (po stronie klienta) przez co jeszcze przez ułamek sekundy wydać nową postać

p.s. daj znać jeśli chciałbyś żebym załatał tego Buga, chętnie dorzuciłbym coś od siebie do projektu

**Comment by dbartoszp** [comment]
Faktycznie było to z nocy z 10.05 na 11.05? Wczoraj właśnie ten błąd naprawiałem (pojawił się inny, którym może zajmę sie dzisiaj XD) 

W tym momencie staleTime dla pobierania aktualnej poprawnej postaci jest inifite, więc teoretycznie dopiero refresh strony powinien zmienić poprawną postać, ale potestuję na devie chociaż z localStorage i datą może być śmiesznie

A co do samej pomocy to na ten moment nie planuję dodatkowych osób dodawać do projektu, bo to taki typowo hobbystyczny i dla nauki ale jeśli by były większe plany to dam znać gdybym potrzebował pomocy

**Comment by RafalKornel** [comment]
Tak, robiłem to 11.05 ok 00:15, więc noc z 10 na 11, ale mogłem mieć starszą wersję kodu pobraną lokalnie :D 

Sprawdzę jeszcze dzisiaj i dam znać. 

BTW propsy za projekt, bardzo fajny mimo że pomysł prosty :D

**Comment by RafalKornel** [comment]
A z innej strony, mógłbym ten błąd rozwiązać trochę inaczej. W `page.tsx` (component chyba Home) jest „use client”, który pobiera ID obecnej postaci i renderuje komponent serwerowy. Ten komponent serwerowy i tak pobiera info o postaci na podstawie tego ID, a więc możesz  od razu wszystko wyrenderowac na serwerze, w tym komponencie serwerowym (czy jakoś tak).

**Comment by dbartoszp** [comment]
Problemu by to raczej nie rozwiązało, bo jeśli mając na myśli komponent serwerowy mówisz o MainGameCard, to przez to że parent (Home) jest komponentem klienta, to automatycznie MainGameCard też jest komponentem klientowym. Mógłbym przenieść pobieranie dzisiejszej correct postaci do MainGameCard, ale nie chcę już tegokomponentu zaśmiecać bo ngl nie jestem dumny bo jest paskudnie duży XD

**Comment by RafalKornel** [comment]
Jeśli wewnątrz MainGameCard uzyjesz useGetCurrentCorrectCharacter zamiast useGetCharacterById to kod ci się uprości bo usuniesz ok 10 linijek w page.tsx, i zmienisz tylko 1 w MainGameCard (no i dodatkowo usuniesz mu propa z ID) 😃

**Comment by dbartoszp** [comment]
W takim wypadku bym musiał lekko zmienić useGetCurrentCharacter (które jak teraz patrzę powinno się nazywać useGetCurrentCharacterId, bo zwraca id a nie całą postać), ale faktycznie mogę tak zrobić  żeby chociaz trochę page.tsx oczyścić xd
